src/utils.js

```diff
@@ -0,0 +1,123 @@
+const fs = require("fs");
+const read = require("fs-readdir-recursive");
+const { v4: uuidv4 } = require("uuid");
+
+function bundleForms() {
+  const formConfigPath = "./formSkeleton/";
+
+  let bundledFormOutput = fs.readFileSync(
+    formConfigPath + "inputFields/input-fields.ttl",
+    "utf8"
+  );
+
+  read(formConfigPath + "forms/").forEach((fileDir, i) => {
+    bundledFormOutput += fs.readFileSync(
+      formConfigPath + "forms/" + fileDir,
+      "utf8"
+    );
+  });
+
+  return bundledFormOutput;
+}
+
+function clearDistFolder() {
+  if (fs.existsSync("./dist")) {
+    fs.rmSync("./dist", { recursive: true });
+  }
+}
+
+function createFormTtlFile({ project, fileName, bundledFormOutput }) {
+  fs.mkdirSync(`./dist/${project.name}/config/semanticForms`, {
+    recursive: true,
+  });
+  fs.writeFileSync(
+    `./dist/${project.name}/config/semanticForms/${fileName}`,
+    bundledFormOutput
+  );
+}
+
+function createResourceMigrationFile({
+  project,
+  timestamp,
+  targetGraph,
+  fileName,
+  fileSize,
+  dateCreated,
+  dateModified,
+}) {
+  fs.mkdirSync(
+    `./dist/${project.name}/config/migrations/${timestamp}-add-new-forms`,
+    {
+      recursive: true,
+    }
+  );
+  fs.writeFileSync(
+    `./dist/${project.name}/config/migrations/${timestamp}-add-new-forms/${timestamp}-add-file-resource.sparql`,
+    generateMigrationContents({
+      targetGraph,
+      fileName,
+      fileSize,
+      dateCreated,
+      dateModified,
+    })
+  );
+}
+
+function generateMigrationContents({
+  targetGraph,
+  fileName,
+  fileSize,
+  dateCreated,
+  dateModified,
+}) {
+  let uuid = uuidv4();
+
+  // TODO: Should we use some sort of deindenting util instead of resetting the indentation here?
+  // prettier-ignore
+  return `\
+PREFIX nfo: <http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#>
+PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX dbpedia: <http://dbpedia.org/resource/>
+
+INSERT DATA{
+  GRAPH <${targetGraph}> {
+    <share://semantic-forms/${fileName}> a nfo:FileDataObject ;
+      dct:type <http://data.lblod.gift/concepts/form-file-type> ;
+      mu:uuid "${uuid}" ;
+      nfo:fileName "${fileName}" ;
+      dct:created "${formatIsoDate(dateCreated)}"^^xsd:dateTime ;
```

Comment:
I'm using the same format as in the [migration](https://github.com/lblod/app-digitaal-loket/blob/ed761a8731ffe8fd51226582f0e6223d460e7f50/config/migrations/20200407100352-automatisch-melding/20200904103600-fix-add-the-forms-file-as-a-file-resource.sparql) that was linked in the readme. Is it a valid dateTime without hour information though?

Reply:
So, interesting question. Did some quick digging. If you look [here](https://www.w3.org/TR/xmlschema11-2/#dateTime), it states:
> **_3.3.7.1 Value Space_**
[dateTime](https://www.w3.org/TR/xmlschema11-2/#dateTime) uses the [date/timeSevenPropertyModel](https://www.w3.org/TR/xmlschema11-2/#dt-dt-7PropMod), **with no properties except [·timezoneOffset·](https://www.w3.org/TR/xmlschema11-2/#vp-dt-timezone) permitted to be absent.**

The way I see that is, all but the timezoneOffset field, should be specified.

Reply:
Ok, I'll add the time information as well. :+1: